modules/videoio/test/test_video_io.cpp

```diff
@@ -77,17 +79,18 @@ class Videoio_Test_Base
         }
         EXPECT_EQ(idx, (int)cap.get(CAP_PROP_POS_FRAMES));
         checkFrameRead(idx, cap);
+#endif
     }
 public:
     void doTest()
     {
-        VideoCapture cap(video_file);
+        VideoCapture cap(video_file, apiPref);
         if (!cap.isOpened())
         {
             std::cout << "SKIP test: Can't open video: " << video_file << std::endl;
             return;
         }
-
+#ifndef HAVE_AVFOUNDATION
```

Comment:
This check should be replaced with dynamic check. See my other comment for details.